builtin/credential/aws-iam/path_role.go

```diff
@@ -0,0 +1,260 @@
+package awsiam
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/fatih/structs"
+	"github.com/hashicorp/vault/helper/policyutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathRole(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "role/" + framework.GenericNameRegex("role"),
+		Fields: map[string]*framework.FieldSchema{
+			"role": {
+				Type:        framework.TypeString,
+				Description: "Name of the role.",
+			},
+			"bound_iam_principal": {
+				Type:        framework.TypeString,
+				Description: "ARN of the IAM principal to bind to this role.",
+			},
+			"ttl": {
+				Type:    framework.TypeDurationSecond,
+				Default: 0,
+				Description: `Duration in seconds after which the issued token should expire. Defaults
+to 0, in whcih case the value will fall back to the system/mount defaults.`,
+			},
+			"max_ttl": {
+				Type:        framework.TypeDurationSecond,
+				Default:     0,
+				Description: "The maximum allowed lifetime of tokens issued using this role",
+			},
+			"policies": {
+				Type:        framework.TypeString,
+				Default:     "default",
```

Comment:
Don't need "default" here as it gets added independently of the backend-defined policies.